IBPSA/Fluid/HeatExchangers/BaseClasses/WetCoilDryRegime.mo

```diff
@@ -0,0 +1,103 @@
+
+  // - water
+  input Modelica.SIunits.ThermalConductance UAWat
+    "UA for water side";
+  input Modelica.SIunits.MassFlowRate mWat_flow
+    "Mass flow rate for water";
+  input Modelica.SIunits.MassFlowRate mWatNonZer_flow(min=Modelica.Constants.eps)
+    "Mass flow rate for water, bounded away from zero";
+
+  input Modelica.SIunits.SpecificHeatCapacity cpWat
+    "Specific heat capacity of water";
+  input Modelica.SIunits.Temperature TWatIn
+    "Water temperature at inlet";
+  // -- air
+  input Modelica.SIunits.ThermalConductance UAAir
+    "UA for air side";
+  input Modelica.SIunits.MassFlowRate mAir_flow(min=Modelica.Constants.eps)
+    "Mass flow rate of air";
+  input Modelica.SIunits.MassFlowRate mAirNonZer_flow(min=Modelica.Constants.eps)
+    "Mass flow rate for air, bounded away from zero";
+  input Modelica.SIunits.SpecificHeatCapacity cpAir
+    "Specific heat capacity of moist air at constant pressure";
+  input Modelica.SIunits.Temperature TAirIn
+    "Temperature of air at inlet";
+  // -- misc.
+  input IBPSA.Fluid.Types.HeatExchangerFlowRegime cfg
+    "The flow regime of the heat exchanger";
+  input Modelica.SIunits.MassFlowRate mAir_flow_nominal
+    "Nominal mass flow rate for air";
+  input Modelica.SIunits.MassFlowRate mWat_flow_nominal
+    "Nominal mass flow rate for water";
+
+  parameter Real delta = 1E-3 "Small value used for smoothing";
+
+  output Modelica.SIunits.HeatFlowRate QTot_flow
+    "Heat transferred from water to air";
+  output Modelica.SIunits.Temperature TWatOut
+    "Temperature of water at outlet";
+  output Modelica.SIunits.Temperature TAirOut
+    "Temperature of air at the outlet";
+  output Real eps(min=0, max=1, unit="1")
+    "Effectiveness for heat exchanger";
+  Modelica.SIunits.ThermalConductance CWat_flow=mWat_flow*cpWat
+    "Capacitance rate of water";
+  Modelica.SIunits.ThermalConductance CAir_flow=mAir_flow*cpAir
+    "Capacitance rate of air";
+  Modelica.SIunits.ThermalConductance CMin_flow_nominal=
+    min(mAir_flow_nominal*cpAir,mWat_flow_nominal*cpWat)
+    "Minimum capacity rate";
+  Modelica.SIunits.ThermalConductance CMax_flow_nominal=
+    max(mAir_flow_nominal*cpAir,mWat_flow_nominal*cpWat)
+    "Maximum capacity rate";
+  Modelica.SIunits.ThermalConductance CMin_flow=
+    IBPSA.Utilities.Math.Functions.smoothMin(
+      x1=CAir_flow,x2=CWat_flow,deltaX=1E-3*(CMax_flow_nominal-CMin_flow_nominal))
+    "Minimum capacity rate";
+  Modelica.SIunits.ThermalConductance UA
+    "Overall heat transfer coefficient";
+  output Modelica.SIunits.Temperature TSurAirOut
+    "Surface Temperature at air outlet";
+equation
+  UA = 1/ (1 / UAAir + 1 / UAWat);
+
+  eps=epsilon_C(
+    UA=UA,
+    C1_flow=CWat_flow,
+    C2_flow=CAir_flow,
+    flowRegime=Integer(cfg),
+    CMin_flow_nominal= CMin_flow_nominal,
+    CMax_flow_nominal=CMax_flow_nominal,
+    delta= delta);
+
+  QTot_flow = eps*CMin_flow*(TAirIn-TWatIn);
+  TAirOut=TAirIn-QTot_flow/(mAirNonZer_flow*cpAir);
```

Comment:
Not sure which media functions you are refering to. Ultimately the changes in the medium properties are made in `IBPSA.Fluid.HeatExchangers.WetCoilEffectivenessNTU` with instances of `IBPSA.Fluid.Humidifiers.Humidifier_u` and `IBPSA.Fluid.HeatExchangers.HeaterCooler_u` based on `QTot_flow` and `mCon_flow`. We do notice some minor discrepancies between the leaving air temperature computed internally by `IBPSA.Fluid.HeatExchangers.BaseClasses.WetCoilWetRegime` and the actual medium temperature at the coil outlet, see plot below. But like you I am not sure whether this may have bad side effects.

![WetCoilWetRegime](https://user-images.githubusercontent.com/10965994/142848291-0b716595-5664-44f7-9132-ea5efd36e807.png)

Reply:
I mean that `h_out = h_in + Q/m_flow` and `T=Medium.Temperature(Medium.setState_phX(p,h,X))` instead of using `cp` directly.

Reply:
I see, thanks for the clarification.
For reference, those temperature variables are used internally to compute the total and sensible heat flow rates and the UA nominal value in fully dry conditions.
I would like to avoid the significant refactoring considering the small impact observed on the temperature evaluation (order of 10-1 K) and the fact that the changes in the medium properties are ultimately applied with the proper medium functions based on `QTot_flow` and `mCon_flow`.
Do you agree to move forward with the current formulation?

Reply:
okay!